apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -225,54 +287,31 @@ const DanceAiModal: React.FunctionComponent<DanceAiProps> = ({onClose}) => {
     }
   };
 
-  const handleUseClick = () => {
-    currentAiModalField?.setValue(resultJson);
-    dispatch(setCurrentAiModalField(undefined));
-  };
-
-  const handleExplanationClick = () => {
-    setMode(Mode.EXPLANATION);
-  };
-
-  const handleStartOverClick = () => {
-    setMode(Mode.SELECT_INPUTS);
-    setInputs([]);
-    setCurrentInputSlot(0);
-    setTypingDone(false);
-    setResultJson('');
-    setShowPreview(false);
-    setProcessingDone(false);
-    setGeneratingNodesDone(false);
-    setGeneratingDone(false);
-  };
-
-  let showConvertButton = false;
-  let showUseButton = false;
-
-  if ((mode === Mode.RESULTS && typingDone) || mode === Mode.RESULTS_FINAL) {
-    if (aiOutput === AiOutput.GENERATED_BLOCKS || aiOutput === AiOutput.BOTH) {
-      showConvertButton = true;
-    }
-    if (aiOutput === AiOutput.AI_BLOCK || aiOutput === AiOutput.BOTH) {
-      showUseButton = true;
-    }
-  }
-
-  const onBlockPreviewDone = useCallback(() => {
-    if (mode === Mode.GENERATING) {
-      setGeneratingDone(true);
-    } else if (mode === Mode.RESULTS) {
-      setShowPreview(true);
-      setMode(Mode.RESULTS_FINAL);
-      setTypingDone(true);
-    }
-  }, [mode, setGeneratingDone, setShowPreview, setTypingDone]);
+  const showUseButton =
+    mode === Mode.RESULTS &&
+    currentToggle === 'ai-block' &&
+    (aiOutput === AiOutput.AI_BLOCK || aiOutput === AiOutput.BOTH);
+
+  const showConvertButton =
+    mode === Mode.RESULTS &&
+    currentToggle === 'code' &&
+    (aiOutput === AiOutput.GENERATED_BLOCKS || aiOutput === AiOutput.BOTH);
+
+  const botImage =
+    mode === Mode.GENERATING &&
+    generatingStep[0] < BAD_GENERATED_RESULTS_COUNT &&
+    generatingStep[1] >= 1
+      ? aiBotNo
+      : mode === Mode.GENERATING && generatingStep[1] >= 1
+      ? aiBotYes
+      : aiBotBorder;
```

Comment:
Took me a bit to parse this :) could this be broken up into something like
```
let botImage = aiBotBorder;
if (mode === Mode.GENERATING && generatingStep[1] >= 1) {
  botImage = generatingStep[0] < BAD_GENERATED_RESULTS_COUNT ? aiBotNo : aiBotYes;
}
```
(assuming the logic is correct?)

Also - could the generating substep just be a boolean flag, since there are only ever two substeps?

Reply:
I'll leave it as a number, but appreciate the simplification, even if it is mutable :)